File: api/inventario-api/DTOs/ProductOutput.cs
namespace inventario_api.DTOs
{
    public class ProductOutput
    {
        public Guid Id { get; set; }
        public required string Name { get; set; }
        public Guid CategoryId { get; set; }
        public string Category { get; set; }
        public decimal Quantity { get; set; }
        public decimal MinimumQuantity { get; set; }
        public DateTime Created { get; set; }
        
    }
}

File: app/inventario_app/lib/models/product_output.dart
class ProductOutput {
  final String id;
  final String name;
  final String categoryId;
  final String category;
  final double quantity;
  final double minimumQuantity;
  final DateTime created;

  ProductOutput({
    required this.id,
    required this.name,
    required this.categoryId,
    required this.category,
    required this.quantity,
    required this.minimumQuantity,
    required this.created,
  });

  factory ProductOutput.fromJson(Map<String, dynamic> json) {
    return ProductOutput(
      id: json['id'],
      name: json['name'],
      categoryId: json['categoryId'],
      category: json['category'],
      quantity: (json['quantity'] as num).toDouble(),
      minimumQuantity: (json['minimumQuantity'] as num).toDouble(),
      created: DateTime.parse(json['created']),
    );
  }
}